packages/opam-test/opam-test.0.1.0/opam

```diff
@@ -11,9 +11,7 @@ build: ["dune" "build" "-p" name "-j" jobs]
 depends: [
   "ocaml" {>= "4.02"}
   "dune" {>= "2.0"}
-  "opam-client"
-  {opam-version >= "2.1" & opam-version < "2.2" & >= "2.1.0" & < "2.2"}
-  "opam-client" {>= "2.2" & < "2.3" & opam-version >= "2.2" & opam-version < "2.3"}
+  "opam-client" {opam-version >= "2.1" & opam-version < "2.2" & >= "2.1.0" & < "2.2"}
```

Comment:
```suggestion
  "opam-client" {opam-version >= "2.1" & opam-version < "2.3" & >= "2.1.0" & < "2.2"}
```

Reply:
Right now it fails with opam-client not found, I think because with the opam-version bounds it is ignoring the constraint (the CI has opam 2.2). Should we try this alternative?

Reply:
Should we make it conflict with opam-version outside the range otherwise?

Reply:
From what i understand, it needs to have the same opam-lib version than the used binary. And there is an api update, it is not compatible with 2.2.x. So maybe have avilability on opam-version=2.1? and kee lib opam-client >= "2.1.0" & < "2.2" ?

note: it was the same issue for opam-build in #24026, cf [failing ci](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/2526f77cadfdab5dfcb2aec30e276da538b17430)

Reply:
```suggestion
  "opam-client" {>= "2.1.0" & < "2.2"}
```